main.go

```diff
@@ -82,6 +108,155 @@ func runMain() error {
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
 
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
 	return nil
 }
+
+func wait() {
+    i := 0
+    for {
+        time.Sleep(time.Second * 1)
+        i++
+    }
+}
+
+// adapted from: https://stackoverflow.com/questions/26709971/could-this-be-more-efficient-in-go
+func grepFile(file string, regexPat string) int64 {
+	r, err := regexp.Compile(regexPat)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", regexPat, err)
+	}
+    patCount := int64(0)
+    f, err := os.Open(file)
+    if err != nil {
+		logrus.Errorf("An error occurred while attempting to open file", err)
+    }
+    defer f.Close()
+    scanner := bufio.NewScanner(f)
+    for scanner.Scan() {
+		if r.Match(scanner.Bytes()) {
+			patCount++
+		}
+    }
+    if err := scanner.Err(); err != nil {
+        logrus.Errorf("An error occurred while scanning file.\n%v", err)
+    }
+    return patCount
+}
+
+func getLatestReleaseVersion(repo *git.Repository, noPrevVersion string) semver.Version {
+	semverRegex, err := regexp.Compile(SEMVER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", SEMVER_REGEX, err)
+	}
+
+	tagrefs, err := repo.Tags()
+	if err != nil {
+		logrus.Errorf("An error occurred while retrieving tags for repository", err)
+	}
+
+	// Trim tagrefs and filter for only tags with X.Y.Z version format
+	var tagSemVers []*semver.Version
+	tagrefs.ForEach(func(tagref *plumbing.Reference) error {
+		tagName := tagref.Name().String()
+		tagName = strings.ReplaceAll(tagName, tagsPrefix, "")
+
+		if semverRegex.Match([]byte(tagName)) {
+			tagSemVer, err := semver.StrictNewVersion(tagName)
+			if err != nil {
+				logrus.Errorf("An error occurred while converting tags to semantic version.", err)
+			}
+			tagSemVers = append(tagSemVers, tagSemVer) 
+		}
+		return nil
+	})
+
+	var latestReleaseTagSemVer *semver.Version
+	if len(tagSemVers) == 0 {
+		latestReleaseTagSemVer, err = semver.StrictNewVersion(NO_PREVIOUS_VERSION)
+		if err != nil {
+			logrus.Errorf("An error occurred while converting tags to semantic version.", err)
+		}
+	} else {
+		sort.Sort(sort.Reverse(semver.Collection(tagSemVers)))
+		latestReleaseTagSemVer = tagSemVers[0]
+	}
+
+	return *latestReleaseTagSemVer
+}
+
+func detectBreakingChanges(changelogFilepath string) bool {
+	changelogFile, err := os.Open(changelogFilepath);
+	if err != nil {
+		logrus.Errorf("Error attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath, err)
+	}
+	defer changelogFile.Close()
+
+	tbdRegex, err := regexp.Compile(TBD_VERSION_HEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", TBD_VERSION_HEADER_REGEX, err)
```

Comment:
1) I'd recommend adding a bit more semantic information, e.g. "Could not parse TBD changelog header regex ......"
2) here and everywhere in the PR, you'll want to be returning an error rather than logging the error